plugins/modules/foreman_search_facts.py

```diff
@@ -125,16 +142,24 @@ def main():
             search=dict(default=""),
             full_details=dict(type='bool', aliases=['info'], default='false'),
             params=dict(type='dict'),
+            organization=dict(),
         ),
     )
 
     module_params = module.clean_params()
     resource = module_params['resource']
     search = module_params['search']
-    params = module_params.get('params')
+    params = module_params.get('params', {})
 
     module.connect()
 
+    if 'organization' in module_params:
+        params['organization_id'] = module.find_resource_by_name('organizations', module_params['organization'], thin=True)['id']
+
+    if params:
+        params = module.foremanapi.resource(resource).action('index').prepare_params(params)
```

Comment:
Can't we move this directly in `list_resource` ?

Reply:
so this will still fail, as it throws away `thin=*` for all resources that don't support it (most), but I think you're right and this doesn't belong into this pr.

Reply:
updated for not doing it in this PR, will open a second one.

@bagasse mind having another look?